controllers/storagecluster/placement.go

```diff
@@ -62,7 +62,7 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookCephv1.Placeme
 
 	topologyKey := getFailureDomain(sc)
 	topologyKey, _ = topologyMap.GetKeyValues(topologyKey)
-	if component == "mon" || component == "mds" || (component == "rgw" && getCephObjectStoreGatewayInstances(sc) > 1) {
+	if (component == "mon" || component == "mds" || (component == "rgw" && getCephObjectStoreGatewayInstances(sc) > 1)) && placement.PodAntiAffinity != nil {
 		if placement.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution != nil {
```

Comment:
```suggestion
		if placement.PodAntiAffinity != nil && placement.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution != nil {
```

might be easier to read

Reply:
changed